chain/txValidator.go

```diff
@@ -264,107 +285,199 @@ func VerifyTransactionWithLedger(txn *transaction.Transaction) error {
 	return nil
 }
 
-type Iterator interface {
-	Iterate(handler func(item *transaction.Transaction) error) error
+type subscription struct {
+	topic      string
+	bucket     uint32
+	subscriber Uint160
```

Comment:
I think we should probably change to pk here instead of Uint160 because sender may want to use some of the subscribers' pk for e2e encryption...

Reply:
Oh nevermind...